### PR TITLE
fix multiple posture submits erroring on top of eachother

### PR DESCRIPTION
**Pull request by andrewpmartinez**
- fix memory leaks on model maps and ziti_error's in posture.c
- avoid using shared memory during individual posture response submits
  for older controllers

### CODE REVIEW

File: library/posture.c [hunk 1]
```diff
@@ -15,80 +15,123 @@ limitations under the License.
...
         uv_unref((uv_handle_t *) &ztx->posture_checks->timer);
     }
 }
 
 extern void ziti_posture_checks_free(struct posture_checks *pcs) {
-    model_map_clear(pcs->previous_responses, NULL);
-    model_map_clear(pcs->current_responses, NULL);
-    FREE(pcs);
-}
+    if (pcs != NULL) {
```
Comment (ekoby) [inline library/posture.c:109]: it is not safe if pcs.timer is not closed

Comment (andrewpmartinez) [reply library/posture.c:109]: Closed? Stopped? 

I was following the pattern I saw in `ziti.c` for `ziti_shutdown()` -> `ziti_ctx_free()`.

Where the `ziti_shutdown` I assumed would always be called first followed by a `ziti_ctx_free`.

Comment (andrewpmartinez) [reply library/posture.c:109]: I can blindly call `uv_timer_stop` here though. It checks the state first to see if it needs stopping.

Comment (andrewpmartinez) [reply library/posture.c:115]: updated

Comment (ekoby) [reply library/posture.c:115]: stop is not enough, I believe, it's still referenced by the loop. you need to call `uv_close()`. freeing memory is not safe until close_cb is called (you can do it inside close_cb)

Comment (andrewpmartinez) [reply library/posture.c:115]: I added `ziti_posture_checks_shutdown()` and call it before anything is freed (one line below this comment string).

Comment (andrewpmartinez) [reply library/posture.c:116]: Fixed?

**Review by ekoby — APPROVED** [review]
LGTM

**Pull request merged**